### ISSUE TITLE
White screen due to large memory usage

### ISSUE THREAD

**Issue by jburrill** [a85/POSTMan-Chrome-Extension#2786]
1. Postman Version: 4.10.3
 2. App (Chrome app or Mac app): App
 3. OS details: win32 6.1.7601 / x64
 4. Is the Interceptor on and enabled in the app: No
 5. Did you encounter this recently, or has this bug always been there: Since today's update
 6. Expected behavior: No disruption to the UI
 7. Console logs (http://blog.getpostman.com/2014/01/27/enabling-chrome-developer-tools-inside-postman/ for the Chrome App, View->Toggle Dev Tools for the Mac app): None available
 8. Screenshots (if applicable)

This seemed to be random.  It has happened repeatedly today.  I have never seen this before today, even after updates were applied.   Today, however, it seems to happened frequently since updating and restarting; usually when selecting an existing query from a collection.

Perhaps a restart of the OS will help.  I have restarted the App each time as nothing repaints; there is only a fully white canvas

**Comment by vegetableman** [comment]
@jburrill Could you share a screenshot of the devtools console and the app?. To open up the console, you could refer: https://www.getpostman.com/docs/errors.

**Comment by jburrill** [comment]
Ummmm.  Sure?  But they are not very descriptive ;-)  Here you go....
![postmanappwhite](https://cloud.githubusercontent.com/assets/19514081/23666288/9642485c-0328-11e7-9991-bab23efba8d7.png)
![postmandevtools](https://cloud.githubusercontent.com/assets/19514081/23666302/9da8706c-0328-11e7-8f6e-b667e5f59d90.png)

**Comment by vegetableman** [comment]
@jburrill Not sure what really could be causing this issue. But you would need to keep the devtools open while you are selecting the collection requests, so that we could see the error logs.

**Comment by jburrill** [comment]
I am not sure what the Dev Tools will tell you.  It is not recording anything. when the white-out occurs.  I also attach the Postman Console output.  But that is not recording any anomalies, either.

![postmanappwhite](https://cloud.githubusercontent.com/assets/19514081/23753761/a573d774-04a8-11e7-8521-11ddd386547c.png)

![postmandevtools](https://cloud.githubusercontent.com/assets/19514081/23753766/a846317c-04a8-11e7-87a2-514cab099a57.png)

![postmanconsole](https://cloud.githubusercontent.com/assets/19514081/23753769/aa9a96f2-04a8-11e7-8dc5-1958a9f0676c.png)

I cleared the dialog on the Developer Tools window to view the content.

**Comment by jburrill** [comment]
Definitely looks like a memory issue.  I ran Task Manager, and one Postman.exe process ran up over 3,000,000K (that's GB) of memory before the app went white.

I should note, that I had not closed out of Postman, but found I could open a New Window from the unresponsive and white app.

**Comment by jburrill** [comment]
I saw the memory rise, again, this time it continued to rise, long after the response was received.  The memory had crept up to over 3GB, again, but fell dramatically to 2GB when I took this screen shot.  It continued to rise a little more, before it went to white.
![postmanprocesses](https://cloud.githubusercontent.com/assets/19514081/23754708/f5c6d8f4-04ab-11e7-9650-5e783e6a26cb.png)

**Comment by vegetableman** [comment]
@jburrill Thanks for sharing the screenshots 👍 . Really appreciate your help on debugging this issue. It seems to be a memory crash scenario.  

I would like to confirm a few more things:
- Did you run any memory/cpu intensive test script?
- Does the crash only happen after sending a request? or does it happen without? (It's highly likely you are facing this issue after sending any kind of request).
- Is any of the response for those requests large?

Furthermore, could you open the app through the command prompt and check for any errors on the prompt after the crash? (example: screenshot below)

![image](https://cloud.githubusercontent.com/assets/1330851/23764436/8365f092-0523-11e7-900d-96d39a12a17e.png)

**Comment by jburrill** [comment]
Very well.   I will tyr that.  But while I am thinking of it, I note that the shortcut I have is not to Postman.exe, directly:
`C:\Users\e207507\AppData\Local\Postman\Update.exe --processStart "Postman.exe"`

**Comment by vegetableman** [comment]
@jburrill In your case, it should be `C:\Users\e207507\AppData\Local\Postman\app-4.10.3\Postman.exe`

**Comment by jburrill** [comment]
Oh, and to answer your questions (while I await what happens from running Postman.exe directly from CLI - as you note above, coincidentlally):

- I am not running any memory intensive script, just standard REST calls
- I believe that the issue happens only after some REST call is made.  But I rarely just open Postman without a purpose.
- The REST responses are often quite large, often 10 records, and with potentially hundreds of attributes for each record.

**Comment by jburrill** [comment]
```Handling of 'wheel' input event was delayed for 3787 ms due to main thread being busy. Consider marking event handler as 'passive' to make the page more responive.```
[sic]

**Comment by jburrill** [comment]
That was in the DevTools Console, this time.  However, there is no trace in the terminal window

common.js:14749

**Comment by vegetableman** [comment]
That's fine. I am just looking for any error messages on the command prompt, after the crash. 
Also, could you let me know the size of those responses? The size value is present right above the response viewer in the rightmost corner (screenshot).

![screen shot 2017-03-10 at 1 37 29 am](https://cloud.githubusercontent.com/assets/1330851/23768489/38631bec-0532-11e7-8b58-b0c059d982d8.png)

**Comment by jburrill** [comment]
There was no trace in the terminal until I manually closed the app, which had gone to white:
```
C:\Users\e207507\AppData\Local\Postman\app-4.10.3>Postman.exe

C:\Users\e207507\AppData\Local\Postman\app-4.10.3>
Closed Window (id: 2)
Closed Window (id: 1)

C:\Users\e207507\AppData\Local\Postman\app-4.10.3>
```
I was not able to retrieve the response for you, as nothing was visible in the app.  I had to start, again.  In each case, there is nothing in the console.
![baseresponse](https://cloud.githubusercontent.com/assets/19514081/23769164/768f30c8-04dc-11e7-8424-67f57ffd0ae5.png)

**Comment by jburrill** [comment]
I was running a second request, of what might be comparable size, and was not able to retrieve the statistics that second time, as the app went white while I was typing :-(.  I have been running such requests for months, with no issue.  It was only after this recent update that the issue showed its head.

**Comment by vegetableman** [comment]
@jburrill The issue seems to be platform-specific, as only few other users have reported a memory crash before. While we investigate it, could you let me know the version number for the earlier working app, you were using (if you remember)?. Also, you could drop a mail to `help@getpostman.com` and we will send you the build for the same.

**Comment by jburrill** [comment]
I am not sure of the earlier version number.  I have been allowing Postman to prompt me to update as they come up.  Would I be far from wrong to assume the earlier version was app-4.10.2?  When I look back at the command-line comments above, my AppData includes both app-4.10.2 and app-4.10.3.

**Comment by jburrill** [comment]
There is a SquirrelSetup.log in AppData/Local/Postman that seems to have version history in it.  4.10.2 is the likely earlier version from what I can see.  That was installed on 2/16/2017, on top of 4.9.3.  With the current 4.10.3 update coming on 3/6/2017.

**Comment by vegetableman** [comment]
@jburrill Thanks for sending the mail. I have replied to it. Do Let me know if you are still facing the issue.

**Comment by redefinitionof** [comment]
I have the same issue on 4.10.3 and 4.10.5 I am running 500 requests that each return 1 MB of data. It seems the large data does influence the occurance. Is it possible to download 4.10.2. I am unable to test due to this.

**Comment by Marnnn** [comment]
I am experiencing exactly the same thing from the updates I installed today (has been notifying me for a while). Request was working fine before.

Postman for Mac
Version 4.10.7
OS X 15.6.0 / x64

I've reverted back to version 4.10.2 and 4.9.3 but these are still showing a white screen some time after loading the request. The request has the following stats; 
Time: 13294ms
Size: 61.2mb

**Comment by ovdsom** [comment]
Hi everybody,
Did anyone manage to make it work?
I am experiencing exactly the same issue of white screen.
Thanks!

Postman for windows
Version 5.0.2
windows x64

More information about the issue:
- There are no error logs visible (It is shown the same image that is present the 9 of March. Blank window which is not recording anything).
- Related to memory usage is Task Manager showing for the app, It is being used 13.1 MB and 177 MB.
- The only app that is opened is POSTMAN in the computer.
- The only request performed is a an authorization token (token request).
- The version that is used is the 5.0.2 (It is the last version for x64 that is present in webpage).

**Comment by SamvelRaja** [comment]
@ovdsom Can you help us in debugging this much further by providing the logs from background layer. To get the logs you need to follow these steps.

Opening up the app from cmd, so that we can get the logs about it.
a. Go to cmd 
b. C:\Users\[USER]\AppData\Local\Postman\app-5.0.0\Postman.exe 
c. This will open the app for the user with console mode to track background data flow.

Once the app is opened kindly send us the logs it leaves when it becomes blank.

**Comment by redefinitionof** [comment]
Hi SamvelRaja,


I have not experienced the issue any more. I expect that it was caused, because my application that I was testing had a memory leak. When all system memory was full, Postman probably couldn't create a UI window any more.


Steven


________________________________
Van: SamvelRaja <notifications@github.com>
Verzonden: donderdag 13 juli 2017 08:13
Aan: postmanlabs/postman-app-support
CC: redefinitionof; Comment
Onderwerp: Re: [postmanlabs/postman-app-support] 4.10.3 White Screen (#2786)


@ovdsom<https://github.com/ovdsom> Can you help us in debugging this much further by providing the logs from background layer. To get the logs you need to follow these steps.

Opening up the app from cmd, so that we can get the logs about it.
a. Go to cmd
b. C:\Users[USER]\AppData\Local\Postman\app-5.0.0\Postman.exe
c. This will open the app for the user with console mode to track background data flow.

Once the app is opened kindly send us the logs it leaves when it becomes blank.

**Comment by redefinitionof** [comment]
OK, I have the issue again: But now immediately. I updated to version v5.1.3 and it worked for 2 days. Now when I start collection runner I see an electron process in task manager increasing to 3GB and at 3GB I get a white screen at the collection runner and the process disappears.

**Comment by redefinitionof** [comment]
There is nothing in the log. Could you explicitly state which logs you mean (how to view them.)? I tried from the postman view menu ->"Show Postman Console::". I also tried memory profile. There are too many calls to make sense out of it.

**Comment by redefinitionof** [comment]
In the allocation profile, there is nothing BTW. The allocation timeline shows all the data, but it seems not from the runner, since I see no major memory leak.

**Comment by reinaldomoreira** [comment]
I'm having the exact same problem when I have a request of 5MB +/- and try to locate any string (CTRL + F)

**Comment by numaanashraf** [comment]
Duplicate of #4751
